### PR TITLE
Add table aws_docdb_cluster Closes #1006

### LINKED ISSUE #1006: Add table aws_docdb_cluster
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.
[AWS Ref](https://docs.aws.amazon.com/documentdb/latest/developerguide/mongo-apis.html)
[Terraform Ref](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/docdb_cluster)
[GO SDK](https://docs.aws.amazon.com/sdk-for-go/api/service/docdb/)

### PR DESCRIPTION
**Pull request by karanpopat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Outputs:

account_id = "111122223333"
aws_partition = "aws"
endpoint = "turbottest64532.cluster-c0e80t0lbvg7.us-east-1.docdb.amazonaws.com"
region_name = "us-east-1"
resource_aka = "arn:aws:rds:us-east-1:111122223333:cluster:turbottest64532"
resource_id = "cluster-LHNWP323HC35LOZWRN7V7JXXS4"
resource_name = "turbottest64532"

Running SQL query: test-get-query.sql
[
  {
    "arn": "arn:aws:rds:us-east-1:111122223333:cluster:turbottest64532",
    "db_cluster_identifier": "turbottest64532",
    "deletion_protection": false,
    "endpoint": "turbottest64532.cluster-c0e80t0lbvg7.us-east-1.docdb.amazonaws.com",
    "engine": "docdb",
    "master_user_name": "turbottest",
    "multi_az": false,
    "storage_encrypted": false,
    "tags_src": [
      {
        "Key": "name",
        "Value": "turbottest64532"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-hydrate-query.sql
[
  {
    "arn": "arn:aws:rds:us-east-1:111122223333:cluster:turbottest64532",
    "db_cluster_identifier": "turbottest64532",
    "deletion_protection": false,
    "endpoint": "turbottest64532.cluster-c0e80t0lbvg7.us-east-1.docdb.amazonaws.com",
    "engine": "docdb",
    "master_user_name": "turbottest",
    "multi_az": false,
    "storage_encrypted": false,
    "tags_src": [
      {
        "Key": "name",
        "Value": "turbottest64532"
      }
    ]
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "arn": "arn:aws:rds:us-east-1:111122223333:cluster:turbottest64532",
    "db_cluster_identifier": "turbottest64532"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql
null
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:rds:us-east-1:111122223333:cluster:turbottest64532"
    ],
    "db_cluster_identifier": "turbottest64532",
    "tags": {
      "name": "turbottest64532"
    },
    "title": "turbottest64532"
  }
]
✔ PASSED

POSTTEST: tests/aws_docdb_cluster

TEARDOWN: tests/aws_docdb_cluster

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  db_cluster_identifier,
  status,
  create_time,
  kms_key_id,
  storage_encrypted
from
  aws_docdb_cluster
where
  not storage_encrypted;
+---------------------------+-----------+---------------------------+------------+-------------------+
| db_cluster_identifier     | status    | create_time               | kms_key_id | storage_encrypted |
+---------------------------+-----------+---------------------------+------------+-------------------+
| docdb-2022-05-16-13-57-40 | available | 2022-05-16T19:29:37+05:30 | <null>     | false             |
+---------------------------+-----------+---------------------------+------------+-------------------+

> select
  db_cluster_identifier,
  member ->> 'DBClusterParameterGroupStatus' as db_cluster_parameter_group_status,
  member ->> 'DBInstanceIdentifier' as db_instance_identifier,
  member ->> 'IsClusterWriter' as is_cluster_writer,
  member ->> 'PromotionTier' as promotion_tier
from
  aws_docdb_cluster
  cross join jsonb_array_elements(members) as member;

+---------------------------+-----------------------------------+---------------------------+-------------------+----------------+
| db_cluster_identifier     | db_cluster_parameter_group_status | db_instance_identifier    | is_cluster_writer | promotion_tier |
+---------------------------+-----------------------------------+---------------------------+-------------------+----------------+
| docdb-2022-05-16-13-57-40 | in-sync                           | docdb-2022-05-16-13-57-40 | true              | 1              |
+---------------------------+-----------------------------------+---------------------------+-------------------+----------------+
```
</details>

### CODE REVIEW

**Comment by rajlearner17** [comment]
Already addressed part of [aws_rds_db_cluster](https://hub.steampipe.io/plugins/turbot/aws/tables/aws_rds_db_cluster)

**Comment by wesky93** [comment]
I don't think docDB can be inquired in aws_rds_cluster table. So why don't we open this PR again?

**Comment by cbruno10** [comment]
@wesky93 I agree, `aws_docdb_cluster` should be a separate table (similar to how TF has a separate resource, [aws_docdb_cluster](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/docdb_cluster) for it).

We'll re-open this PR and look to merge it in when ready.

**Comment by cbruno10** [comment]
@karanpopat Can you please resolve merge conflicts?

Also, please see the changes we did to the `aws_neptune_db_cluster` and `aws_rds_db_cluster` tables around engine filtering + table doc updates, to ensure that we're only getting DocDB clusters in this table:
- https://github.com/turbot/steampipe-plugin-aws/commit/8f82fee1fa738e3e96cdf0efbbaad326270505e9
- https://github.com/turbot/steampipe-plugin-aws/commit/bdf88ee47d8440f77a10464086a1eae6bcf0dc8b

File: aws/service_v2.go [hunk 1]
```diff
@@ -81,6 +82,14 @@ func CostExplorerClient(ctx context.Context, d *plugin.QueryData) (*costexplorer
 	return costexplorer.NewFromConfig(*cfg), nil
 }
 
+func DocDBService(ctx context.Context, d *plugin.QueryData) (*docdb.Client, error) {
```
Comment (cbruno10) [inline aws/service_v2.go:85]:
```suggestion
func DocDBClient(ctx context.Context, d *plugin.QueryData) (*docdb.Client, error) {
```

This function name should be consistent with the others

File: aws/table_aws_docdb_cluster.go [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+	"github.com/aws/aws-sdk-go-v2/service/docdb/types"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsDocDBCluster(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_docdb_cluster",
+		Description: "AWS Doc DB Cluster",
```
Comment (cbruno10) [inline aws/table_aws_docdb_cluster.go:21]:
```suggestion
		Description: "AWS DocumentDB Cluster",
```

File: aws/table_aws_docdb_cluster.go [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_docdb_cluster.listDocDBClusters", "api_error", err)
+			return nil, err
+		}
+
+		for _, cluster := range output.DBClusters {
+
```
Comment (cbruno10) [inline aws/table_aws_docdb_cluster.go:275]:
```suggestion
```

File: aws/table_aws_docdb_cluster.go [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+			if helpers.StringSliceContains(
```
Comment (cbruno10) [inline aws/table_aws_docdb_cluster.go:276]: In Neptune and RDS, we needed this as non-Neptune/RDS DB clusters would be returned.

I'm assuming it's the same case here, but if you have any Neptune DB or RDS clusters in the same region, does the DocDB `DescribeDBClusters` function also return those?

Comment (cbruno10) [inline aws/table_aws_docdb_cluster.go:276]:
```suggestion
			// The DescribeDBClusters API returns non-DocDB clusters as well,
			// but we only want DocDB clusters here.
			if helpers.StringSliceContains(
```

Only add this comment if non-DocDB clusters are returned too by default

File: docs/tables/aws_docdb_cluster.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
+# Table: aws_docdb_cluster
+
+Amazon DocumentDB (with MongoDB compatibility) is a fast, reliable, and fully managed database service. Amazon DocumentDB makes it easy to set up, operate, and scale MongoDB-compatible databases in the cloud. With Amazon DocumentDB, you can run the same application code and use the same drivers and tools that you use with MongoDB.
+
+## Examples
+
```
Comment (cbruno10) [inline docs/tables/aws_docdb_cluster.md:6]: Please add a `Basic info` here too, like we have in other docs.

File: docs/tables/aws_docdb_cluster.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+where
+  backup_retention_period > 7;
+```
+
+### Get avalability zone count for each cluster
+
+```sql
+select
+  db_cluster_identifier,
+  jsonb_array_length(availability_zones) availability_zones_count
```
Comment (cbruno10) [inline docs/tables/aws_docdb_cluster.md:39]:
```suggestion
  jsonb_array_length(availability_zones) as availability_zones_count
```

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@karanpopat Please see review comments, thanks!

File: aws/table_aws_docdb_cluster.go (hunk 4 above)
Comment (karanpopat) [reply aws/table_aws_docdb_cluster.go:276]: Yes, I checked that the RDS clusters are being returned by the `DescribeDBClusters` function.

**Pull request merged**